File: v131016/TeeNew/Chart_Clipboard.pas
unit Chart_Clipboard;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series;

type
  TChartClipboard = class(TBaseForm)
    Button1: TButton;
    Series1: TPointSeries;
    Series2: TLineSeries;
    Series3: TFastLineSeries;
    procedure Button1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TChartClipboard.Button1Click(Sender: TObject);
begin
  Chart1.CopyToClipboardBitmap;

  // custom size:
  // Chart1.CopyToClipboardBitmap( Rect( 0,0, 400, 200 ) );
end;

procedure TChartClipboard.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.SeriesList.FillSampleValues(10);
end;

initialization
  RegisterClass(TChartClipboard);
end.

File: TeeNew/Builder2010/Chart_Clipboard.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Chart_Clipboard.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TChartClipboard *ChartClipboard;
//---------------------------------------------------------------------------
__fastcall TChartClipboard::TChartClipboard(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TChartClipboard::Button1Click(TObject *Sender)
{
  Chart1->CopyToClipboardBitmap();

  // custom size:
  // Chart1.CopyToClipboardBitmap( Rect( 0,0, 400, 200 ) );
}
//---------------------------------------------------------------------------

void __fastcall TChartClipboard::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(10);
  Series2->FillSampleValues(10);
  Series3->FillSampleValues(10);
}
//---------------------------------------------------------------------------

